crates/wasi-http/wit/deps/http/types.wit

```diff
@@ -175,11 +176,13 @@ interface types {
     ///
     /// This method consumes the `response-outparam` to ensure that it is
     /// called at most once. If it is never called, the implementation
-    /// will respond with an error.
+    /// will respond with an error. If there is an error sending the request,
+    /// the error result will contain both a description of the error and the
+    /// `response-outparam`, allowing the `set` operation to be retried.
```

Comment:
If we reject invalid headers eagerly, then I think there shouldn't be any error possible here.  In particular, the `set` operation shouldn't be describing the actual sending of the response over the network (that kind of error would be reflecting in the streaming operations for writing to and finishing the response body); this operation should be the equivalent of `*outparam = result;`.  Or am I missing something else that can happen here?

Reply:
Good point. Switching to having forbidden headers rejected at `fields.set` and `fields.append` removes the need to allow validation here.